contentgrid-appserver-query-engine-impl-jooq/src/test/java/com/contentgrid/appserver/query/engine/jooq/EventsDispatchTest.java

```diff
@@ -0,0 +1,878 @@
+                updateEventConsumer
+        )).isInstanceOf(SomeEventFailureException.class);
+
+        // Transaction rolled back → it's not updated in db
+        var existing = queryEngine.findById(APPLICATION, created.getIdentity().toRequest(), PERMIT_ALWAYS).orElseThrow();
+        var attr = (SimpleAttributeData<?>) existing.getAttributeByName(ATTRIBUTE_A.getName()).orElseThrow();
+        assertThat(attr.getValue()).isEqualTo(123L);
+    }
+
+    @Test
+    void verifyUpdate_dbThrows() {
+        queryEngine.create(
+                APPLICATION,
+                EntityCreateData.builder()
+                        .entityName(ENTITY_A.getName())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_A.getName())
+                                .value(123)
+                                .build())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_B.getName())
+                                .value(true)
+                                .build())
+                        .build(),
+                PERMIT_ALWAYS,
+                createEventConsumer
+        );
+
+        var second = queryEngine.create(
+                APPLICATION,
+                EntityCreateData.builder()
+                        .entityName(ENTITY_A.getName())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_A.getName())
+                                .value(456)
+                                .build())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_B.getName())
+                                .value(false)
+                                .build())
+                        .build(),
+                PERMIT_ALWAYS,
+                createEventConsumer
+        );
+
+        // Update second entity to violate unique constraint
+        var updatePayload = EntityData.builder()
+                .name(ENTITY_A.getName())
+                .id(second.getId())
+                .attribute(SimpleAttributeData.builder()
+                        .name(ATTRIBUTE_A.getName())
+                        .value(456)
+                        .build())
+                .attribute(SimpleAttributeData.builder()
+                        .name(ATTRIBUTE_B.getName())
+                        .value(true)
+                        .build())
+                .build();
+        assertThatThrownBy(() -> queryEngine.update(
+                APPLICATION,
+                updatePayload,
+                PERMIT_ALWAYS,
+                updateEventConsumer
+        )).isInstanceOf(QueryEngineException.class);
+
+        // Failure to update → no events sent
+        Mockito.verify(updateEventConsumer, Mockito.never()).onEntityUpdate(any(), any(), any());
+    }
+
+    @Test
+    void verifyDelete_eventThrows() {
```

Comment:
Missing a `verifyDelete_dbThrows()`, which would have caught the issue that I mentionned earlier.

Yes, it is possible for the database to refuse a delete. For example, when it's referenced by a foreign key that is not nullable.